api/ncf_api_flask_app/views.py

```diff
@@ -74,59 +83,89 @@ def get_path_from_args (request):
 @check_auth
 def get_techniques():
   """Get all techniques from ncf folder passed as parameter, or default ncf folder if not defined"""
-  # We need to get path from url params, if not present put "" as default value
-  path = get_path_from_args(request)
-
-  techniques = ncf.get_all_techniques_metadata(alt_path = path)
-  resp = jsonify( techniques )
-  return resp
+  try:
+    # We need to get path from url params, if not present put "" as default value
+    path = get_path_from_args(request)
+  
+    techniques = ncf.get_all_techniques_metadata(alt_path = path)
+    resp = jsonify( techniques )
+    return resp
+  except :
+    errstr = traceback.format_exc()
+    sys.stderr.write(errstr)
+    sys.stderr.flush()
+    error = jsonify ({ "error": "Unknown internal error", "details": errstr})
+    error.status_code = 500
+    return  error
 
 @app.route('/api/generic_methods', methods = ['GET'])
 @check_auth
 def get_generic_methods():
   """Get all generic methods from ncf folder passed as parameter, or default ncf folder if not defined"""
-  # We need to get path from url params, if not present put "" as default value
-  path = get_path_from_args(request)
-
-  generic_methods = ncf.get_all_generic_methods_metadata(alt_path = path)
-  resp = jsonify( generic_methods )
-  return resp
+  try:
+    # We need to get path from url params, if not present put "" as default value
+    path = get_path_from_args(request)
+  
+    generic_methods = ncf.get_all_generic_methods_metadata(alt_path = path)
+    resp = jsonify( generic_methods )
+    return resp
+  except :
+    errstr = traceback.format_exc()
+    sys.stderr.write(errstr)
+    sys.stderr.flush()
+    error = jsonify ({ "error": "Unknown internal error", "details": errstr})
```

Comment:
Mayhe the error could be "Unknown internal error while fetching generic methods"